python/ray/dashboard/head.py

```diff
@@ -96,6 +97,12 @@ def __init__(
         self._modules_to_load = modules_to_load
         self._modules_loaded = False
 
+        self._thread_pool_executor = ThreadPoolExecutor(
+            # We limit number of workers to avoid starving the event loop thread.
```

Comment:
let's explain a bit more why too many worker threads will starve the event loop thread